python/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK.py

```diff
@@ -0,0 +1,399 @@
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK
+Description:
+  Check whether the Amazon EBS snapshots are not publicly restorable. The rule is NON_COMPLIANT if the RestorableByUserIds field is set to 'all'.
+Trigger:
+  Periodic
+Reports on:
+  AWS::::Account
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+    Given: No snapshots with RestorableByUserIds parameter set to 'all'
+     Then: Return COMPLIANT
+  Scenario: 2
+    Given: One or more snapshots with RestorableByUserIds parameter set to 'all'
+     Then: Return NON_COMPLIANT with Annotation containing SnapshotIDs
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+
+# Function to get IDs for all non compliant resources
+def generate_snapshot_id_list(snapshots, event):
+    snapshot_ids = []
+    for snapshot in snapshots:
+        snapshot_ids.append(snapshot['SnapshotId'])
+    return snapshot_ids
+
+
+# Function to obtain all public Amazon EBS snapshots
+def get_public_snapshots(ec2_client, owner_id):
+    snapshots = []
+    next_token = None
+    snapshots_result = {}
+    while True:
+        if not next_token:
+            snapshots_result = ec2_client.describe_snapshots(Filters=[{'Name':'owner-id', 'Values':[owner_id]}], OwnerIds=[owner_id], RestorableByUserIds=['all'], MaxResults=1000)
+        else:
+            snapshots_result = ec2_client.describe_snapshots(Filters=[{'Name':'owner-id', 'Values':[owner_id]}], OwnerIds=[owner_id], RestorableByUserIds=['all'], NextToken=next_token, MaxResults=1000)
+        snapshots.extend(snapshots_result['Snapshots'])
+        if 'NextToken' in snapshots_result:
+            next_token = snapshots_result['NextToken']
+        else:
+            return snapshots
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+    ec2_client = get_client('ec2', event)
+    public_snapshots_result = get_public_snapshots(ec2_client, event['accountId'])
+    if not public_snapshots_result:
+        return build_evaluation(event['accountId'], 'COMPLIANT', event)
+    snapshot_ids = generate_snapshot_id_list(public_snapshots_result, event)
+    for snapshot_id in snapshot_ids:
+        evaluations.append(build_evaluation(snapshot_id, 'NON_COMPLIANT', event, annotation='Public Amazon EBS Snapshot: {}'.format(snapshot_id)))

```

Comment:
The change was not requested. Keep accountId as a resource_id. The question asked was about the lengh of the annotation.